src/app/pages/HomePage/ParaTimeSelector/index.tsx

```diff
@@ -71,41 +81,95 @@ const ZoomOutBtnFade = styled(Fade)(() => ({
   transitionDelay: '500ms !important',
 }))
 
+const QuickPinchZoomOuter = styled('div')(() => ({
+  '> div': {
+    position: 'absolute',
+    inset: 0,
+  },
+}))
+
+const QuickPinchZoomInner = styled('div')(() => ({
+  position: 'absolute',
+  width: '100%',
+  height: '100%',
+}))
+
 const ParaTimeSelectorCmp: FC<ParaTimeSelectorProps> = ({ disabled }) => {
+  const graphRef = useRef<SVGSVGElement & HTMLElement>(null)
+  const quickPinchZoomRef = useRef<PinchZoom>(null)
+  const quickPinchZoomInnerRef = useRef<HTMLDivElement>(null)
+  const theme = useTheme()
+  const isMobile = useMediaQuery(theme.breakpoints.down('sm'))
   const { t } = useTranslation()
   const exploreBtnTextTranslated = t('home.exploreBtnText')
 
-  const [step, setStep] = useState<ParaTimeSelectorStep>(ParaTimeSelectorStep.ENABLE_EXPLORE)
-  const [selectedGraphEndpoint, setSelectedGraphEndpoint] = useState<GraphEndpoint>(GraphEndpoint.CONSENSUS)
+  const [step, setStep] = useState<ParaTimeSelectorStep>(ParaTimeSelectorStep.EnableExplore)
+  const [selectedGraphEndpoint, setSelectedGraphEndpoint] = useState<GraphEndpoint>()
+
+  const { width, height } = useResizeObserver<SVGSVGElement>({
+    ref: graphRef,
+  })
+
+  useEffect(() => {
+    if (selectedGraphEndpoint) {
+      quickPinchZoomRef.current?.scaleTo(GraphUtils.getScaleTo(selectedGraphEndpoint, { width, height }))
+    }
+  }, [selectedGraphEndpoint])
 
   const onExploreClick = () => {
-    setStep(ParaTimeSelectorStep.EXPLORE)
+    setStep(ParaTimeSelectorStep.Explore)
   }
 
   const onZoomOutClick = () => {
-    setSelectedGraphEndpoint(GraphEndpoint.CONSENSUS)
+    setSelectedGraphEndpoint(GraphEndpoint.Consensus)
   }
 
+  const onUpdate = ({ x, y, scale }: UpdateAction) => {
+    if (!isMobile) {
+      return
+    }
```

Comment:
I would name this "onPinchZoom" and remove the `if (!isMobile)`

Reply:
Right, this function won't be called on mobile. Good catch 👍